code/modules/chemistry/tools/ampoules.dm

```diff
@@ -42,6 +42,47 @@
 	playsound(user.loc, "sound/impact_sounds/Generic_Snap_1.ogg", 50, 1)
 	return
 
+// based off of plants.dm
+/obj/item/reagent_containers/ampoule/attackby(obj/item/W as obj, mob/user as mob)
+	if (istype(W, /obj/item/spacecash) || istype(W, /obj/item/paper))
+		var/obj/item/clothing/mask/cigarette/custom/ampoule/C = new(user.loc)
+		C.reagents.maximum_volume = (src.reagents.total_volume)
+		src.force_drop(user)
+		src.set_loc(C)
+		C.ampoule = src
+		src.reagents.copy_to(C.reagents)
+
+		boutput(user, "<span class='alert'>You wrap the [src] in the [W].</span>")
+
+		W.force_drop(user)
+		qdel(W)
+		user.put_in_hand_or_drop(C)
+
+	else if (istype(W, /obj/item/bluntwrap))
+		var/obj/item/bluntwrap/B = W
+		var/obj/item/clothing/mask/cigarette/cigarillo/ampoule/doink = new(user.loc)
+		src.force_drop(user)
+		src.set_loc(doink)
+		doink.ampoule = src
+		doink.reagents.clear_reagents()
+		src.reagents.copy_to(doink.reagents)
+
+		if(B.flavor)
+			doink.flavor = B.flavor
+
+		W.reagents.trans_to(doink, W.reagents.total_volume)
+		doink.reagents.maximum_volume = (src.reagents.total_volume + 50)
+		doink.name = "[reagent_id_to_name(doink.flavor)]-flavored [pick("doink","'Rillo","cigarillo","brumbpo")]"
+
+		boutput(user, "<span class='alert'>You roll the [src] in the [W] and make a fat doink.</span>")
+
+		W.force_drop(user)
+		qdel(W)
+		user.put_in_hand_or_drop(doink)
+
```

Comment:
You may want a final else condition here for saftey's sake

Reply:
Done (: